sdk/log/logger_bench_test.go

```diff
@@ -85,3 +87,25 @@ func newTestLogger(t testing.TB) log.Logger {
 	)
 	return provider.Logger(t.Name())
 }
+
+func BenchmarkLoggerRetrieval(b *testing.B) {
+	b.Run("Create different loggers single-threaded", func(b *testing.B) {
+		provider := NewLoggerProvider()
+
+		b.ReportAllocs()
+		for n := 0; n < b.N; n++ {
+			provider.Logger(fmt.Sprintf("test-%d", n))
+		}
+	})
+
+	b.Run("Create different loggers parallel", func(b *testing.B) {
+		provider := NewLoggerProvider()
+
+		b.ReportAllocs()
```

Comment:
```suggestion
		b.ReportAllocs()
		b.ResetTimer()
```

So creating the provider doesn't count towards the benchmark.